File: hello-mcp-ts/mcp-client/test_client.ts
import { HelloClient } from "./client.js";

async function runAllTests(port: number) {
  const client = new HelloClient(`http://localhost:${port}`);

  try {
    console.log("=== 测试1: 列举Hello MCP工具 ===");
    const tools = await client.listTools();
    console.log(`\n列举到的工具:\n${tools}\n`);

    console.log("=== 测试2: 测试Hello MCP - 按名称查询 ===");
    const result = await client.getElement("氢");
    console.log(`查询氢元素结果: ${result}\n`);

    console.log("=== 测试3: 测试MCP工具调用 - 按位置查询 ===");
    const result2 = await client.getElementByPosition(6);
    console.log(`查询原子序数为6的元素结果: ${result2}\n`);
  } finally {
    await client.close();
  }
}

// 解析命令行参数
const args = process.argv.slice(2);
let port = 3000;

for (let i = 0; i < args.length; i++) {
  if (args[i] === "--port" && i + 1 < args.length) {
    port = parseInt(args[i + 1], 10);
  }
}

console.log(`连接到 MCP 服务器: http://localhost:${port}\n`);
runAllTests(port).catch(console.error);

File: hello-mcp-csharp/mcp-client/TestClient.cs
using Microsoft.Extensions.Logging;

namespace Feuyeux.AI.Hello.Client;

/// <summary>
/// MCP 客户端测试程序
/// </summary>
public class TestClient
{
    public static async Task<int> Main(string[] args)
    {
        int port = 9900;

        // 解析命令行参数
        for (int i = 0; i < args.Length; i++)
        {
            if (args[i] == "--port" && i + 1 < args.Length)
            {
                port = int.Parse(args[i + 1]);
            }
        }

        using var loggerFactory = LoggerFactory.Create(builder =>
        {
            builder.AddConsole();
            builder.SetMinimumLevel(LogLevel.Information);
        });

        var logger = loggerFactory.CreateLogger<HelloClient>();
        var client = new HelloClient(logger, $"http://localhost:{port}");

        Console.WriteLine($"连接到 MCP 服务器: http://localhost:{port}");
        await RunAllTests(client);

        return 0;
    }

    private static async Task RunAllTests(HelloClient client)
    {
        await TestListTools(client);
        await TestGetElementByName(client);
        await TestGetElementByPosition(client);
    }

    /// <summary>
    /// 测试1: 列举Hello MCP工具
    /// </summary>
    private static async Task TestListTools(HelloClient client)
    {
        Console.WriteLine("\n=== 测试1: 列举Hello MCP工具 ===");
        var tools = await client.ListTools();
        Console.WriteLine($"\n列举到的工具:\n{tools}\n");
    }

    /// <summary>
    /// 测试2: 测试Hello MCP - 按名称查询
    /// </summary>
    private static async Task TestGetElementByName(HelloClient client)
    {
        Console.WriteLine("=== 测试2: 测试Hello MCP - 按名称查询 ===");
        var result = await client.GetElement("氢");
        Console.WriteLine($"查询氢元素结果: {result}\n");
    }

    /// <summary>
    /// 测试3: 测试MCP工具调用 - 按位置查询
    /// </summary>
    private static async Task TestGetElementByPosition(HelloClient client)
    {
        Console.WriteLine("=== 测试3: 测试MCP工具调用 - 按位置查询 ===");
        var result = await client.GetElementByPosition(6);
        Console.WriteLine($"查询原子序数为6的元素结果: {result}\n");
    }
}
